enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,432 @@
+CVO filtering, such as turning off a particular api for the cluster.
+
+### Non-Goals
+
+* Making control-plane critical components optional (k8s apiserver, openshift apiserver, openshift controller,
+  networking, etc)
+* Defining which components should be disable-able (this will be up to component teams to classify themselves
+as `capabilities` or not)
+* Providing a way to install OLM operators as part of the initial cluster install.  This EP is about making
+the install experience around the existing CVO-based components more flexible, not adding new components to the
+install experience.
+* Allowing components to be disabled post-install.
+* Eliminating or replacing cluster profiles
+* Encoding logic in the installer itself about which components should be disabled under specific circumstances
+
+
+## Proposal
+
+### User Stories
+
+* As a user creating a new cluster that will be managed programmatically, I do not want the additional
+security exposure and resource overhead of running the web console.  I would like a way to install
+a cluster that has no console out of the box, rather than having to disable it post-install or
+modify rendered manifests in a way that requires deep understanding of the OCP components/resources.
+
+* As a team scaffolding a managed service based on openshift, I want to minimize the footprint of my
+clusters to the components I need for the service.
+
+* As a user creating a cluster that will never run an image registry, I do not want the additional overhead
+of running the image registry operator, or have to remove the default registry that is created.
+
+* As a team using openshift for a specific use case such as edge deployments, I want to provide
+an install experience that disables components that aren't needed for my use case.
+
+### API Extensions
+
+N/A - no api extensions are being introduced in this proposal
+
+### Implementation Details/Notes/Constraints [optional]
+
+The CVO already has the ability to respect annotations on resources, as can be seen
+[here](https://github.com/openshift/cluster-kube-apiserver-operator/blob/c03c9edf5fddf4e3fb1bc6d7afcd2a2284ca03d8/manifests/0000_20_kube-apiserver-operator_06_deployment.yaml#L10) and leveraged [here](https://github.com/openshift/hypershift/blob/b98bc43467a73921891ede7d6477757b647f1985/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L152-L153).
+This proposal consists of two parts:
+
+1) Formalizing a concept of a "capability" annotation which allows a given resource to be excluded based
+on installer input. For example the console related resources could be annotated as
+
+```yaml
+annotations:
+  capability.openshift.io/console: "true"
+```
+
+Note: "true" means "included by default unless explicitly excluded", in the future we may enable capabilities
+to mark themselves as "false" meaning "excluded by default, unless explicitly included".
```

Comment:
Hmm, how does the current proposal's `spec.capabilities.default` in ClusterVersion interact with these docs?  Does `default: exclude` count as "explicitly excluding" the component?  I'd expect it does.  But if so, the only way to fall back to a component's preferred default would be to leave `spec.capabilities.default` unset.  Is it expected to be a pointer property?

Reply:
this note should be removed, it pre-dated the concept of spec.capabilities.default.

Reply:
(technically it says we could allow "false" in the future, but if we did we'd have to define how that interacts w/ the spec.capabilities.default)

Reply:
did we want `spec.capabilities.default` to be a pointer to leave that door open?  Or are we confident enough to close the door on per-component default in the first round of this?

Reply:
I guess the door will be open to `spec.capabilites.default: perComponentDefaults` or whatever, even if we don't go with a pointer, so :shrug: